pkg/logql/log/exec_test.go

```diff
@@ -0,0 +1,281 @@
+// Copyright 2010 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
```

Comment:
This is just a fork of https://github.com/golang/go/blob/master/src/regexp/exec_test.go. I'm not certain we can merge this due to the licensing.

Reply:
I think it is allowed, but I wouldn't bother - one or two examples where you check the string output of `allNonGreedy` seems fine as a test; you don't need to re-test the whole regex package.